Sweeper/View Controllers/PinsMapViewController.swift

```diff
@@ -35,27 +37,49 @@ class PinsMapViewController: UIViewController {
         mapView.delegate = self
         mapView.isHidden = true
         view.addSubview(mapView)
-        loadPins()
         setupLocationButton()
         registerForNotifications()
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // refilter on every load in case pins were visited on the details page
+        pins = pins.filter { (pin: Pin) -> Bool in
+            let visited = pin.visited ?? false
+            if visited {
+                let idx = self.annotations.index(where: { (annotation: PinAnnotation) -> Bool in
+                    return annotation.pin == pin
+                })
+                
+                if let idx = idx {
```

Comment:
This looks like an n^2 search. We can probably get rid of the pinsArray and only maintain an array of pinannotations. Although, I think for our demo purposes this isn't too bad.

Reply:
Yea it's kinda gross. We are tracking too much data on this page.